spark/src/main/java/org/apache/iceberg/spark/source/BaseDataReader.java

```diff
@@ -88,7 +118,74 @@ public T get() {
     return current;
   }
 
-  abstract Iterator<T> open(FileScanTask task);
+  abstract Pair<Schema, Iterator<T>> getJoinedSchemaAndIteratorWithIdentityPartition(
```

Comment:
Since this is part of the Base Readers api, would be good to add a docstring on what this is supposed to do and where it's being used.